src/App.tsx

```diff
@@ -1,25 +1,43 @@
-import React from 'react';
+/* eslint-disable max-len */
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items = getNumbers(1, 42);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState<string>('5');
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const total = items.length;
+  const pages = Math.ceil(total / +perPage);
+
+  const startIndex = (currentPage - 1) * +perPage;
+  const endIndex = startIndex + +perPage;
```

Comment:
You are converting 'perPage' to a number multiple times which is not efficient. Consider converting it once and storing it in a variable or ensuring that 'perPage' is always a number.